control/xferfcn.py

```diff
@@ -468,6 +468,82 @@ def __str__(self, var=None):
 
         return outstr
 
+    def to_zpk(self, var=None):
+        """Return string representation of the transfer function as factorized
+        polynomials.
+
+        Examples
+        --------
+        >>> from control import tf
+        >>> G = tf([1],[1, -2, 1])
+        >>> G.to_zpk()
+               1
+        ---------------
+        (s - 1) (s - 1)
+
+        """
+
+        mimo = self.ninputs > 1 or self.noutputs > 1
```

Comment:
Not sure if it would work as well with the way you have written your logic but there is a `self.issiso()` method available.

Reply:
Ah, better. Will fix.